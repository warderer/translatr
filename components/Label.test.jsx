import { it, describe, expect } from 'vitest';
import { render } from '@testing-library/react'; // Me permite renderizar componentes con un dom virtual en consola
import Label from './Label';

describe('[ components / Label ]', () => {
    describe('when Label is rendered', () => {
        it('should render a label', () => {
            // Test de Componente Visual Estructural (De Entrada y Salida)
            // Patrón de Testing AAA

            // ARRANGE: Es la parte de la preparacion de los datos
            const props = {
                count: 0,
                maxCount: 100,
            }

            // ACT: Es la accion que se va a ejecutar
            // asFragment es un snapshot de lo que se renderizo sin un contenedor
            const { asFragment } = render(<Label {...props} />);

            // ASSERT: Es la parte de la verificación de los datos
            // La primera vez que corre el test, se crea un snapshot de lo que se renderizo, la segunda vez que corre el test, se compara el snapshot con lo que se renderizo y si son iguales, el test pasa, si son diferentes, el test falla
            expect(asFragment()).toMatchSnapshot();
        });
    });
});