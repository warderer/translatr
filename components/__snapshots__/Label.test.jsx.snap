// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[ components / Label ] > when Label is rendered > should render a label 1`] = `
<DocumentFragment>
  <div
    class="h-6 text-right mt-2 text-sm text-gray-500 dark:text-gray-300"
  >
    0 / 100
  </div>
</DocumentFragment>
`;
